pkg/tasks/queue/pgqueue.go

```diff
@@ -710,11 +710,13 @@ func (p *PgQueue) RemoveAllTasks() error {
 	return nil
 }
 
-func (p *PgQueue) ListenForCancel(ctx context.Context, taskID uuid.UUID, cancelFunc context.CancelFunc) {
+func (p *PgQueue) ListenForCancel(ctx context.Context, taskID uuid.UUID, cancelFunc context.CancelCauseFunc) {
 	logger := zerolog.Ctx(ctx)
 	conn, err := p.Pool.Acquire(ctx)
 	if err != nil {
-		logger.Error().Err(err).Msg("ListenForCancel: error acquiring connection")
+		if !errors.Is(ErrNotRunning, context.Cause(ctx)) {
```

Comment:
You might add a comment to explain "sometimes a task can finish before the Listen is initiated" for future self (or me)

Reply:
added a comment